### PR TITLE
App listing2

### PR DESCRIPTION
**Pull request by sri2k1us**
- Add Type Filter
- Add `Browse All Apps`
- Add 302 redirection logic for HPC Auth
- switch to `axios` for Terrain API calls.

Note about `axios` switch:
I had to switch to `axios` cos `node-fetch 2.6.0` wouldn't return `Location` headers for `302` redirects and `node-fetch 3.x beta` worked well for `302` redirect but returns incorrect results by ignoring query params.
      
image

- HPC auth redirect

image

- Note this PR is on top https://github.com/cyverse-de/sonora/pull/77. So reviews can start from https://github.com/cyverse-de/sonora/pull/80/commits/768cc5a9872f81591dea14625a35b843fe5e908a

### CODE REVIEW

File: src/server/api/terrain.js [hunk 1]
```diff
@@ -106,5 +116,7 @@ export const call = (
         };
     }
 
-    return fetch(apiURL, requestOptions);
+    requestOptions.url = apiURL.toString();
+
+    return axios(requestOptions);
```
Comment (psarando) [inline src/server/api/terrain.js:121]: Will changing this from `fetch` to `axios` break the `fetchMock` calls in the stories?

File: src/components/apps/AppNavigation.js [hunk 2]
```diff
@@ -161,7 +203,46 @@ function AppNavigation(props) {
...
+                        intl,
+                        "categoriesMenuItemAriaControl"
+                    )}
+                    aria-label={allAppsCategory.name}
+                >
+                    <ListItemIcon>{allAppsCategory.icon}</ListItemIcon>
+                    <ListItemText>{allAppsCategory.name}</ListItemText>
+                </ListItem>
             </Menu>
+            <div className={classes.divider} />
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/components/apps/AppNavigation.js:231]: The `Divider` component places vertical or horizontal line. I think this is just pushing off all the elements to the right.

File: src/server/api/terrain.js (hunk 1 above)
Comment (sri2k1us) [reply src/server/api/terrain.js:121]: Good catch! Test seems to be broken but I think I found a `axios` mock lib - https://www.npmjs.com/package/axios-mock-adapter. I will update the tests.

File: package.json [hunk 3]
```diff
@@ -66,7 +67,7 @@
     "js-yaml": "^3.13.1",
     "keycloak-connect": "^7.0.1",
     "next": "^9.3.1",
-    "node-fetch": "^2.6.0",
+    "node-fetch": "^3.0.0-beta.4",
```
Comment (ash3rz) [inline package.json:70]: Are we completely switched over at this point to axios?  Do we still need this dependency?

File: src/components/apps/AgaveAuthPromptDialog.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        window.location.replace(location);
+        handleClose();
+    };
+    return (
+        <Dialog
+            open={open}
+            onClose={handleClose}
+            id={build(baseId, ids.AGAVE_AUTH_PROMPT_DIALOG)}
+            fullScreen={fullScreen}
+            aria-labelledby={formatMessage(intl, "agaveRedirectTitle")}
```
Comment (ash3rz) [inline src/components/apps/AgaveAuthPromptDialog.js:35]: `aria-labelledby` should contain an element ID, I believe.  This looks to be a shared message for the dialog title.  So in your case, you just have to add an ID to the `DialogTitle` component and add that same ID here in the `aria-labelledby`.

File: package.json (hunk 3 above)
Comment (sri2k1us) [reply package.json:70]: I will double check uploads one last time and then I can pull the plug on `node-fetch`

**Comment by sri2k1us** [comment]
Looks like we still need `node-fetch` at least until we port `fetch` call in `CyverseAppBar` which I already did in AppDetails PR #83 .

Thanks for the reviews.

**Pull request merged**